esp-preempt/src/run_queue.rs

```diff
@@ -71,7 +72,39 @@ impl RunQueue {
         let current_prio = self.ready_priority.ready();
         debug!("pop - from level: {}", current_prio);
 
-        let popped = self.ready_tasks[current_prio].pop();
+        cfg_if::cfg_if! {
+            if #[cfg(multi_core)] {
+                use esp_hal::system::Cpu;
+
+                let other = match Cpu::current() {
+                    Cpu::AppCpu => Cpu::ProCpu,
+                    Cpu::ProCpu => Cpu::AppCpu,
+                };
+
+                // Look iteratively through priority levels - this will ensure we can find a task even if all high priority tasks are pinned to the other CPU.
```

Comment:
There has to be a better way to do this, like tracking the priority of the core-pinned tasks separately, but I'll take "working in any capacity" right now.